ddtrace/internal/runtime/__init__.py

```diff
@@ -23,38 +25,47 @@ def _generate_runtime_id():
 _RUNTIME_ID = _generate_runtime_id()
 
 
-if hasattr(os, "register_at_fork"):
+def ddtrace_at_fork():
+    global _RUNTIME_ID
+
+    _RUNTIME_ID = _generate_runtime_id()
+    _rand.seed()
```

Comment:
This has nothing to do here.

I'm all for consolidation, but the mechanism deserves its own file and API, not being stuffed there.

Reply:
Hmm. IMO it's okay to have all fork-affected features in a single place within the library. "runtime" makes sense to me as the place to put it. Having one function also makes sense for it for users who fork from C-code and would need to manually reset the library state - hence the name `ddtrace_at_fork`.

However, I do agree that it's not clear or intuitive to add another feature to be run/updated at fork. It's tricky due to the timing inconsistency in the ability to execute hooks when fork hooks aren't available. The API would have to be something like: register a "fork-safe" function with an optional hook to be run when a fork has occurred, before running the function.

I have a rough idea of what this could look like, I'll investigate.